plugins/kuadrant-backend/src/router.ts

```diff
@@ -182,41 +183,60 @@ export async function createRouter({
       const { userId } = await getUserIdentity(req, httpAuth, userInfo);
       const apiProduct = req.body;
       const namespace = apiProduct.metadata?.namespace;
-      const planPolicyRef = apiProduct.spec?.planPolicyRef;
+      const targetRef = apiProduct.spec?.targetRef;
 
       if (!namespace) {
         throw new InputError('namespace is required in metadata');
       }
 
-      if (!planPolicyRef?.name || !planPolicyRef?.namespace) {
-        throw new InputError('planPolicyRef with name and namespace is required');
+      if (!targetRef?.name || !targetRef?.kind) {
+        throw new InputError('targetRef with name and kind is required');
       }
 
-      // fetch the planpolicy to get plan details
-      const planPolicy = await k8sClient.getCustomResource(
-        'extensions.kuadrant.io',
-        'v1alpha1',
-        planPolicyRef.namespace,
-        'planpolicies',
-        planPolicyRef.name,
-      );
-
-      // extract plans from planpolicy
-      const plans = planPolicy.spec?.plans || [];
-
-      if (plans.length === 0) {
-        throw new InputError('selected planpolicy has no plans defined');
-      }
-
-      // inject plans into apiproduct spec
-      apiProduct.spec.plans = plans;
-
       // set the owner to the authenticated user
       if (!apiProduct.spec.contact) {
         apiProduct.spec.contact = {};
       }
       apiProduct.spec.contact.team = `user:default/${userId}`;
 
+      // temporary: populate plans from planpolicy until controller implements this
```

Comment:
this all gets nuked when https://github.com/Kuadrant/kuadrant-backstage-plugin/issues/35 lands - just kept it here so that some sort of planpolicy syncing exists for demoing for the next week or two